### PR TITLE
Shibboleth: Reduce nesting and remove duplication

### PR DESCRIPTION
**Pull request by jrchamp**
I was stepping through this code and I found it a bit confusing. To help myself understand:

- I treated all `return` as leaving the code and removed the `else` nesting that just mean the subroutine continued processing. 
- I moved code that could only be reached when a statement was true (i.e. not false) to the block that checked that the statement was true.
- I noticed 3 variables were being assigned twice and removed the copies that were being overwritten 100% of the time.
- I removed a duplicate document comment that did not reflect the code that it was commenting.

Please note that I'm not primarily a Perl programmer, so please make sure these changes make sense. Thank you for taking the time to review these changes. I am primarily intending to improve readability and remove unused code.

### CODE REVIEW

**Comment by jrchamp** [comment]
On a side note: Is there a real reason that only MD5 is supported? Unless the user field has a 32 character limit, it seems like SHA-1 through SHA-512 would be viable options. [1]

[1] https://perldoc.perl.org/Digest.html

**Comment by drgrice1** [comment]
This is set to merge into the master branch.  Only hotfixes should go into master.  I am rather certain this does not qualify as a hotfix.

File: lib/WeBWorK/Authen/Shibboleth.pm [hunk 1]
```diff
@@ -63,57 +62,50 @@ sub get_credentials {
...
-		#    not to rely on internal logins if there's a check_user
-		#    failure.
-		$self->{external_auth} = 1;
+	debug("Shib is on!");
 
-		if ( $r->param("user") && ! $r->param("force_passwd_authen") ) {
-			return $self->SUPER::get_credentials( @_ );
-		}
-		
-		if ( defined ($ENV{$ce->{shibboleth}{session_header}}) && defined( $ENV{$ce->{shibboleth}{mapping}{user_id}} ) ) {
```
Comment (drgrice1) [inline lib/WeBWorK/Authen/Shibboleth.pm:78]: This re-introduces the incorrect $ENV usage fixed in PR#1056 (also on line 80 below).

Comment (jrchamp) [reply lib/WeBWorK/Authen/Shibboleth.pm:78]: #1056 appears to be a fix, but was not applied to master. My commit does not substantively change those lines, only the indentation.

**Comment by jrchamp** [comment]
> This is set to merge into the master branch. Only hotfixes should go into master. I am rather certain this does not qualify as a hotfix.

Thank you for reviewing this, I'll see if I can rebase this on the develop branch.

**Comment by jrchamp** [comment]
Okay, rebased on develop. Sorry for the branch confusion.

**Comment by mgage** [comment]
Thank you @jrchamp for the improvements we appreciate your contribution.  And thank you @glennricster for the review.

**Pull request merged**